### ISSUE TITLE
USER ISSUE: Dach - roof

### ISSUE THREAD

**Issue by TimoNuernberg** [StrangeLoopGames/EcoIssues#19323]
\
**Version:** 0.9.1.4 beta release-116

**Steps to Reproduce:**
Dach bauen, mit Roof und Roof-fixes Aussehen; 

**Expected behavior:**
die blöcke clippen korrekt an die Nachbarn an 

**Actual behavior:**
die blöcke reagieren nicht auf die Nachbar blöcke; es verhält sich so, als wäre kein Dach-Block daneben

**Do you have mods installed? Does the issue happen when no mods are installed?:**
Nein

**Please provide log files. Instruction is here https://github.com/StrangeLoopGames/EcoIssues/wiki/Getting-the-Eco-game-log-file
![image](https://user-images.githubusercontent.com/63196305/99153625-c9afc400-26a9-11eb-8d94-65728f24d5e9.png)
![image](https://user-images.githubusercontent.com/63196305/99153633-d46a5900-26a9-11eb-8327-d59952ce0635.png)

**Comment by StallEF** [comment]
different types of wood do not connect how roof blocks

**Comment by SlayksWood** [comment]
I guess this is another problem which prevents Roof from connecting to other new forms. For example:
- select Roof Form and build square:
![image](https://user-images.githubusercontent.com/45708377/99222287-4ac6a280-27f3-11eb-80fb-06b61859f5a4.png)
- select Roof side and add line:
![image](https://user-images.githubusercontent.com/45708377/99222518-cf192580-27f3-11eb-9754-7aef79dfc378.png)
- try to place roof corner here:
![image](https://user-images.githubusercontent.com/45708377/99222533-db9d7e00-27f3-11eb-969e-05dcac046c8f.png)
- roof will not connected with Roof corner:
![image](https://user-images.githubusercontent.com/45708377/99222618-fcfe6a00-27f3-11eb-84ce-896c5423d201.png)
- remove the unconnected Roof block and replace it with Roof Side:
![image](https://user-images.githubusercontent.com/45708377/99222716-2f0fcc00-27f4-11eb-88ca-284b79a6aca4.png)

So the problem is that you need build only with Roof or only with others Roof blocks.
@keegan-o Is it possible to do something about this?

**Comment by keegan-o** [comment]
Not really. Roof connects to itself, the whole point of other blocks is that they are 100% customizable. Making roof blocks see and connect to them as well is not feasible. I think ultimately we need a more advanced kind of placement/fill type so that we can fully deprecate and remove the 'roof' forms.

**Comment by StallEF** [comment]
@keegan-o  roof block worked on Eco 8 but have bug on Eco 9
![2020-12-30](https://user-images.githubusercontent.com/26339332/110513253-131f9580-8117-11eb-9681-7eb19d629767.png)
![2021-01-01](https://user-images.githubusercontent.com/26339332/110513312-229ede80-8117-11eb-8670-e60f48f24e25.png)
